llvm/unittests/Support/KnownBitsTest.cpp

```diff
@@ -537,19 +537,13 @@ TEST(KnownBitsTest, BinaryExhaustive) {
       [](const KnownBits &Known1, const KnownBits &Known2) {
         return KnownBits::mulhs(Known1, Known2);
       },
-      [](const APInt &N1, const APInt &N2) {
-        unsigned Bits = N1.getBitWidth();
-        return (N1.sext(2 * Bits) * N2.sext(2 * Bits)).extractBits(Bits, Bits);
-      },
+      [](const APInt &N1, const APInt &N2) { return APIntOps::mulhs(N1, N2); },
```

Comment:
Could these be:
```suggestion
      &APIntOps::mulhs,
```
?

Reply:
Hm... I think it should because the first one uses a lambda function as a functor, while the second one directly passes a pointer to the member function APIntOps::mulhs.